File: backend/schemas/status.py
"""
schemas/status.py — Status response models.
"""

from typing import Literal

from pydantic import BaseModel


class ServiceStatus(BaseModel):
    name: str
    status: Literal["healthy", "degraded", "down"]
    latency_ms: int | None = None


class StatusOut(BaseModel):
    api: Literal["healthy", "degraded"]
    services: list[ServiceStatus]
    uptime_seconds: int
    version: str

File: src/api/status.ts
/**
 * src/api/status.ts
 *
 * System health status API calls.
 *
 * Backend route: GET /status
 */

import { apiGet } from './client';

export interface BackendServiceStatus {
  name: string;
  status: 'healthy' | 'degraded' | 'down';
  latency_ms: number | null;
}

export interface BackendStatusOut {
  api: 'healthy' | 'degraded';
  services: BackendServiceStatus[];
  uptime_seconds: number;
  version: string;
}

export async function getSystemStatus(): Promise<BackendStatusOut> {
  return apiGet<BackendStatusOut>('/status');
}
